### DISCUSSION TITLE
Data Summary

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#356] [Component suggestions]
<img width="480" alt="visuele weergave van het data summary component" src="https://github.com/user-attachments/assets/92ac1527-dbb1-4e55-af91-1656b753d6e9">

We willen graag meer leren over het **Data Summary** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1706798708688949)

---

## Naam

Data Summary

## Doel

Lijst met eigenschappen en bijbehorende data over een onderwerp.

## Ook bekend als

Data List

## Gerelateerde componenten

[Description List](https://github.com/orgs/nl-design-system/discussions/216), [Form Summary](https://github.com/orgs/nl-design-system/discussions/215)

## Onderzoek

**Bold of Regular**
We zijn op zoek naar een rationale achter de keuze voor Bold of Regular voor de 'Key' of 'Value'. Over het algemeen lijkt de 'Key' bold te worden weergegeven en de 'Value' regular. Echter kan het omgekeerde ook voorkomen (Amsterdam, DSO).

Bij Amsterdam lijkt hier nog geen definitieve richtlijn voor te zijn. [Zie comment van Nico Druif in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1706694925029049?thread_ts=1705579174.146329&cid=C7KDQRXMZ)

Vragen rondom dit punt werden ook gesteld op UX stackexchange:
- [When displaying statistics to a user, is it better to bold the data or the label?](https://ux.stackexchange.com/questions/7938/when-displaying-statistics-to-a-user-is-it-better-to-bold-the-data-or-the-label)
- [Description and Value, which should be bold?](https://ux.stackexchange.com/questions/56754/description-and-value-which-should-be-bold)

Daar wordt o.a. het volgende benoemd:

> So a rule of thumb:
> Long lists (>5 items) seen rarely (<10 a month): Bold the label.
> Short lists (<5 items) seen frequently (>10 a month): Bold the information

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-descriptionlist--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/DescriptionList)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=8487-27946&mode=design&t=hExV10m61JMlev2n-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1238-4479&mode=design&t=KR6lmR3Gjt3oZuqW-4)

---

### Screenshots / Varianten

**Default**
![Scherm­afbeelding 2024-02-01 om 16 46 09](https://github.com/nl-design-system/backlog/assets/545207/b1169c88-0ae3-403f-818f-891e35728192)

**With caption**
![Scherm­afbeelding 2024-02-01 om 16 48 21](https://github.com/nl-design-system/backlog/assets/545207/769db278-877e-41ef-999d-0cf7bff3a27f)


---

### Notities
- Onder de naam: Description list én Data list.
- HTML wel op basis van dl, dt, en dd.
- Documentatie beschikbaar voor Content guidelines en Do's and Dont's.
- In Figma de optie om horizontaal en verticaal uit te lijnen.
- Bij een small viewport wordt de lijst verticaal getoond.

**Update:** Op basis van gebruik verplaatst van [discussion voor het Form Summary component](https://github.com/orgs/nl-design-system/discussions/215) naar deze discussion.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- [Documentatie website](https://dso-toolkit.nl/62.8.4/components/lists/definition-list)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-definition-list--default&viewMode=story)

---

### Screenshots / Varianten

**Default**
![Scherm­afbeelding 2024-02-01 om 17 03 38](https://github.com/nl-design-system/backlog/assets/545207/6ff2ba0a-a8a6-48b9-9e7e-276c655c3a1c)

**Vertical**
![Scherm­afbeelding 2024-02-01 om 17 03 46](https://github.com/nl-design-system/backlog/assets/545207/8a989ece-0326-4102-9d1e-67891448d13b)

**Emphasize description**
![Scherm­afbeelding 2024-02-01 om 17 03 55](https://github.com/nl-design-system/backlog/assets/545207/b22a0ff0-4e97-4457-94ef-9ff570504a55)

**Vertical  emphasize description**
![Scherm­afbeelding 2024-02-01 om 17 04 02](https://github.com/nl-design-system/backlog/assets/545207/93b9e2a1-b2a0-4db4-a8ac-12b66c960545)

**Bordered**
![Scherm­afbeelding 2024-02-01 om 17 04 40](https://github.com/nl-design-system/backlog/assets/545207/bf159289-3857-4b95-a93b-2b9d0dba5172)

---

### Notities
- Onder de naam: Definition list
- HTML wel op basis van dl, dt, en dd.
- Richtlijnen voor gebruik
- Veel opties
  - Horizontal of verticaal uitlijnen
  - Nadruk op term of beschrijving
  - Verdelingen over kolommen
  - Borders of niet
- Bij een small viewport wordt de lijst verticaal getoond.

**Update:** Op basis van gebruik verplaatst van [discussion voor het Description list component](https://github.com/orgs/nl-design-system/discussions/216) naar deze discussion.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/definition-list)

---

### Screenshots / Varianten

**Default**
![Scherm­afbeelding 2024-02-01 om 17 15 59](https://github.com/nl-design-system/backlog/assets/545207/1b217c50-7aa9-4d9f-b161-95597832ba47)

**Horizontaal**
![Scherm­afbeelding 2024-02-01 om 17 16 08](https://github.com/nl-design-system/backlog/assets/545207/8dfc9df3-89a6-4db0-9440-7a4d453267a2)

**Met lange titel**
![Scherm­afbeelding 2024-02-01 om 17 16 14](https://github.com/nl-design-system/backlog/assets/545207/91ce1d11-9fcc-4c00-9139-938da52413a7)

**Collapsible definition list**
![Scherm­afbeelding 2024-02-01 om 17 16 20](https://github.com/nl-design-system/backlog/assets/545207/e0321423-0db3-46b9-8f61-1eccecc81d53)


---

### Notities
- Onder de naam: Definition list
- Documentatie over wanneer wel en niet te gebruiken.
- Documentatie per variant.
- Bij een small viewport wordt de lijst verticaal getoond.


**Update:** Op basis van gebruik verplaatst van [discussion voor het Form Summary component](https://github.com/orgs/nl-design-system/discussions/215) naar deze discussion. In de documentatie van DUO wordt wel expliciet genoemd dat dit component ook gebruikt kan worden aan het eind van een formulier om antwoorden te controleren en mogelijk wijzigen. Zie ook [het template 'Overzichtspagina'](https://uno.dfront.rijkscloud.nl/#/patronen/formulier-overzichtspagina). Daarom een verwijzing naar deze comment opgenomen in [discussion voor de Form Summary component](https://github.com/orgs/nl-design-system/discussions/215).

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

- [Documentatie website](https://componenten.lenteveld.nl/openingstijden-locatie)
- [Patternlab](https://patternlab.lenteveld.nl/?p=viewall-molecules-OpeningHours)

---

### Screenshots / Varianten

**Openingstijden**
![Scherm­afbeelding 2024-02-05 om 11 37 45](https://github.com/nl-design-system/backlog/assets/545207/093b86de-9334-48a6-9ee3-5a2c4b143d06)

---

### Notities
- Voor het component: Openingstijden en locatie
- HTML wel op basis van dl, dt, en dd.
- Bij een small viewport wordt de lijst verticaal getoond.

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-data-list--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/data-list)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4196-18718&mode=design&t=gmL7S2xsKHZzc92X-4)

---

### Screenshots / Varianten

**Data list**
![Scherm­afbeelding 2024-02-05 om 12 06 51](https://github.com/nl-design-system/backlog/assets/545207/32b9a0b3-e1f2-4e01-b4be-91261fa8a5df)

---

### Notities
- Onder de naam: Data list.
- HTML wel op basis van dl, dt, en dd.
- Bij een small viewport wordt de lijst verticaal getoond.

**Comment by jeffreylauwers** [comment]
### Amsterdam (interne applicaties)

Comment van [Mike Alders](https://github.com/MikeClickr) verplaatst van [discussion voor het Description list component](https://github.com/orgs/nl-design-system/discussions/216) naar deze discussion.

> Ik ben voorstander dat labels regular font krijgen en de waarden/values bold font.
> Rationale: de waarden/values zijn belangrijkste voor de gebruiker om te scannen.
> 
> Nadeel is dat i.g.v. combinatie van inputvelden en description list inconsistentie kan optreden. In Amsterdam zijn de labels van inputs juist bold. Echter, in de meeste situaties worden data lists los van een formulier getoond.
> Zie bijgaande geanonimiseerde voorbeelden.

![Datalist Amsterdam - Desktop](https://github.com/nl-design-system/backlog/assets/6008601/6608cbec-8408-4956-b1c4-ec384980c0b4)
![Datalist Amsterdam - Mobiel](https://github.com/nl-design-system/backlog/assets/6008601/46e3da1d-907a-48bf-81c5-01972044c85d)

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-data-list--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/data-list)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=1238-4479&mode=design&t=Ieh3n67lII9xiCUj-4)

---

### Screenshots / Varianten

**Default**
![Scherm­afbeelding 2024-02-05 om 12 20 36](https://github.com/nl-design-system/backlog/assets/545207/470a7afd-768c-47b7-ba19-6d0d3d776a93)

**Action link**
![Scherm­afbeelding 2024-02-05 om 12 20 41](https://github.com/nl-design-system/backlog/assets/545207/de2fc7e0-b7ad-484a-9b82-0721c36d38d1)

**Action button**
![Scherm­afbeelding 2024-02-05 om 12 20 45](https://github.com/nl-design-system/backlog/assets/545207/746338eb-5a72-4c65-8125-b411e3f323a7)

---

### Notities
- Onder de naam: Data list.
- HTML wel op basis van dl, dt, en dd.
- Documentatie beschikbaar over:
  - wanneer te gebruiken;
  - verschillende soorten data optimaal weer te geven;
  - vertalen van data
  - rtl toepassing
  
**Update:** Op basis van gebruik verplaatst van [discussion voor het Form Summary component](https://github.com/orgs/nl-design-system/discussions/215) naar deze discussion. In de documentatie (Storybook) staat het volgende:

> De data list component gebruik je onder andere voor:
> 
> - een overzicht van gegevens uit een formulier, in een controle-stap voor het verzenden
> - een overzicht vooringevulde gegevens, aan het begin van een formulier
> - een overzicht van bekende gegevens, bijvoorbeeld in een profiel-pagina

Daarom een verwijzing naar deze comment opgenomen in [discussion voor de Form Summary component](https://github.com/orgs/nl-design-system/discussions/215).

**Comment by jeffreylauwers** [comment]
Dit component wordt ook wel eens Data list genoemd. Maar de naam Datalist wordt in HTML al gebruikt voor iets anders...

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

**Reply by RubenSibon** [reply]
Dit is eigenlijk de enige, bijna uitsluitende, reden om niet voor Data List te gaan. Anders zou volgens mij en @VincentSmedinga Data List een betere naam zijn dan Data Summary:

1. Het woord list sluit aan bij het HTML element dat gebruikt wordt: `dl`, i.e. “description list”;
2. Het verschil met een tabel wordt beter benadrukt: de waarden in een kolom van een tabel moet van hetzelfde soort/type zijn om bijvoorbeeld sorteren en filteren mogelijk te maken; een list mag verschillende soorten waarden na elkaar opnoemen.

Extra argumenten voor Data List:

- Utrecht [noemt het component ook Data List](https://github.com/nl-design-system/utrecht/tree/main/components/data-list);
- Den Haag noemt het [in de `src` bestanden Description List](https://github.com/nl-design-system/denhaag/tree/main/components/DescriptionList/src), maar in de [Readme heet het ook Data List](https://github.com/nl-design-system/denhaag/blob/main/components/DescriptionList/README.md#data-list).

N.b.: interessant is wel dat voor zowel Utrecht als Den Haag het component eigenlijk dé Description List is en dat er geen aparte Data of Form Summary componenten lijken te bestaan.

**Comment by jeffreylauwers** [comment]
Tijdens de Open Dag van september de naamgeving van de componenten Summary List en Description List besproken. Uiteindelijk zijn we uitgekomen op [Form Summary](https://github.com/orgs/nl-design-system/discussions/215), Data Summary en [Description List](https://github.com/orgs/nl-design-system/discussions/216).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Data Summary op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=data+summary)

**Comment by bas-kroese** [comment]
Tijdens de NLDS Design Open Hour van 11 aug 2026 kwam dit onderwerp aan bod:

Bij Amsterdam hebben we ervoor gekozen om component '**Data Summary**' aan te bieden, waarmee direct ook het patroon form summary kan worden gemaakt. 

Vooralsnog alleen in Figma beschikbaar: https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=48983-138

Wanneer op een pagina een data summary moet worden gemaakt met meerdere groepen (bijvoorbeeld meerdere toegevoegde personen in een formulier) dan kiezen we er vooralsnog voor om het op deze manier op te lossen op de pagina:
<img width="713" height="1249" alt="image" src="https://github.com/user-attachments/assets/42e948a0-3ece-4edf-be4d-fa8a4724b20d" />

Redenering:

- Meer flexibiliteit in de opmaak (risico op te veel variatie in de uitwerking zou je kunnen afdekken met goede documentatie en patronen)
- Voor screenreaders ook prima in te richten. Dat word al lastiger waarschijnlijk als je het binnen een component wil afdekken.

**Comment by RubenSibon** [comment]
Voor Amsterdam zijn we nu bezig met het Data Summary-component.

Daarvoor heb ik nog eens gekeken naar de drie NLDS-pagina’s rond **Description List**, **Data Summary** en **Form Summary**, plus naar de community-implementaties die daar nu aan gekoppeld zijn.

Mijn belangrijkste conclusie is dat Form Summary geen toegevoegde waarde heeft als apart component, maar vooral een toepassing/patroon van Data Summary is. In een formulier- of data entryflow gebruik je in wezen dezelfde bouwstenen: een lijst van labels en waarden, eventueel met acties zoals “Wijzigen” en eventueel gegroepeerd per onderwerp (zie ook [Data Summary in onze Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=48983-138)). De extra context zit dan eerder in omliggende componenten en paginapatronen dan in een fundamenteel andere componentstructuur.

Daarnaast valt de huidige naamgeving en mapping op:

- Bij **Description List** [noemt Den Haag het component](https://github.com/nl-design-system/denhaag/tree/main/components/DescriptionList) in de readme **Data List**, terwijl het [technisch nog een description list](https://github.com/nl-design-system/denhaag/tree/main/components/DescriptionList/src) is.

- Bij **Data Summary** [staat Utrecht met een implementatie](https://github.com/nl-design-system/utrecht/tree/main/components/data-list) die **Data List** heet, maar onder water ook een HTML description list gebruikt.

- Bij **Form Summary** staat opnieuw Utrecht met diezelfde implementatie, terwijl die dus ook al aan Data Summary gekoppeld is.

Daardoor ontstaat nu het beeld dat drie NLDS-componentpagina’s deels naar **dezelfde soort implementaties** verwijzen, maar onder verschillende namen en in verschillende stadia van voltooidheid.

Ik zou daarom het volgende voorstellen:

1. **Form Summary samenvoegen met Data Summary**.

    Het verschil voelt niet groot genoeg om twee aparte componenten te rechtvaardigen. Een form summary is dan gewoon een specifieke toepassing van dezelfde component, bijvoorbeeld als patroon onder Pages of templates.

    Ik kan me voorstellen dat een component terugnemen, zeker als die al op “Community” staat, geen optie is. En om het erger te maken: Data Summary staat nog op “Help Wanted”.

2. **Data Summary** hernoemen naar **Data List**.

    Die naam heeft duidelijk precedent: Utrecht gebruikt hem al, en Den Haag noemt in de documentatie dus ook zo.

    Er is echter wel een conflict met HTML `<datalist>` element. Een lastige kwestie dus, maar ik vind het wel bespreken waard.

3. Suggestie voor de NLDS website: de **community-implementaties beter koppelen aan de juiste componentpagina’s**, zodat de documentatie beter uitlegt wat nu precies waarin thuishoort. Het lijkt mij momenteel niet helemaal te kloppen.
